File: Backend/SOP/SOP/Entities/PartGroup.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SOP.Entities
{
    public class PartGroup
    {
        [Key]
        public int Id { get; set; }

        [Column(TypeName = "nvarchar(255)")]
        public string PartName { get; set; }

        [Column(TypeName = "decimal(12,2)")]
        public decimal Price { get; set; }

        [Column(TypeName = "nvarchar(255)")]
        public string Manufacturer { get; set; }

        [Column(TypeName = "nvarchar(255)")]
        public string WarrantyPeriod { get; set; }

        [Column(TypeName = "datetime")]
        public DateTime ReleaseDate { get; set; }

        [Column(TypeName = "int")]
        public int Quantity { get; set; }

        [ForeignKey("PartType.Id")]
        public int PartTypeId {get; set;}

        public PartType PartType { get; set;}
    }
}

File: Frontend/src/app/models/partGroup.ts
import { PartType } from "./partType";

/*
* This interface represents the PartGroup model.
* It contains the properties of the PartGroup model.
*/
export interface PartGroup {
    id: number,
    partName: string,
    price: number,
    manufacturer: string,
    warrantyPeriod: string,
    releaseDate: Date,
    quantity: number,
    partTypeId: number,
    //* get the partType object from the partType.ts file
    partType: PartType,
}
